src/Tree/HydroTree.cpp

```diff
@@ -1449,17 +1449,63 @@ void HydroTree<ndim,ParticleType>::FindMpiTransferParticles
     Box<ndim>& nodebox = mpinodes[inode].domain;
 
     // Start from root-cell
-    int NumPartFound = tree->FindBoxOverlapParticles(nodebox, all_particles_to_export,
+    tree->FindBoxOverlapParticles(nodebox, particles_to_export[inode],
                                                      hydro->GetParticleArray()) ;
 
     // Copy particles to per processor array
-    if (NumPartFound > 0)
-      particles_to_export[inode].insert(particles_to_export[inode].end(),
-                                        all_particles_to_export.end() - NumPartFound,
-                                        all_particles_to_export.end());
+    all_particles_to_export.insert(all_particles_to_export.end(),
+                                        particles_to_export[inode].begin(),
+                                        particles_to_export[inode].end());
   }
   //-----------------------------------------------------------------------------------------------
 
+#ifndef NDEBUG
+  VerifyUniqueIds(all_particles_to_export.size(),hydro->Nhydro,&all_particles_to_export[0]);
```

Comment:
We're struggling with Travis with having enough time on travis as it is, so verify all it going to be difficult with travis. But yes, open an issue. We should discuss it at least.